src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -231,10 +231,17 @@ tf_image_augmenter::prepare_augmented_images(
 
     for (size_t i = 0; i < aug_annotations.size(); i++) {
       pybind11::buffer_info buf_ann = aug_annotations[i].request();
-      turi::neural_net::shared_float_array annotate =
-          turi::neural_net::shared_float_array::copy(
-              static_cast<float*>(buf_ann.ptr),
-              std::vector<size_t>(buf_ann.shape.begin(), buf_ann.shape.end()));
+      size_t num_annotations = buf_ann.shape[0];
+      turi::neural_net::shared_float_array annotate;
+      if (num_annotations > 0) {
+        annotate = turi::neural_net::shared_float_array::copy(
+            static_cast<float*>(buf_ann.ptr),
+            std::vector<size_t>(buf_ann.shape.begin(), buf_ann.shape.end()));
+      } else {
+        std::vector<float> data = {0};
+        std::vector<size_t> shape = {1};
+        annotate = turi::neural_net::shared_float_array::wrap(data, shape);
```

Comment:
Why can't we use the `wrap` method on the `shared_float_array` that wraps a float, i.e. https://github.com/apple/turicreate/blob/master/src/ml/neural_net/float_array.hpp#L140 ?

Reply:
okay!